### PR TITLE
support for Repeat function

### LINKED ISSUE #24398: Support parsing more functions SQL
In the process of investigating #24053, I found that there are still many functions that are not supported by the parser engine, and I think we can improve it gradually.

Here are some references:
#### MySQL
- Document: https://dev.mysql.com/doc/refman/8.0/en/functions.html
- PR: #24396

####
- [x] REPEAT https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_repeat
- [x] EXTRACT https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_extract
- [ ] CAST https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast
- [ ] Optimize MySQLStatementSQLVisitor#visitCastFunction

### PR DESCRIPTION
**Pull request by kanha-gupta**
Fixes #24398 .

Changes proposed in this pull request:
  
support for REPEAT function

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by kanha-gupta** [comment]
@RaigorJiang @strongduanmu Please guide if the approach is fine ? if yes then i'll proceed with adding support for others too.

**Review by strongduanmu — CHANGES_REQUESTED** [review]
Hi @kanha-gupta, please remove useless `|` after `REPEAT`.

File: test/it/parser/src/main/resources/case/dml/select-expression.xml [hunk 1]
```diff
@@ -2089,6 +2089,23 @@
         </projections>
     </select>
 
+    <select sql-case-id="select_with_repeat_function">
+      <projections start-index="7" stop-index="25">
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select-expression.xml:2093]: Please keep the same indent as the other cases.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by complone — APPROVED** [review]

**Review by strongduanmu — APPROVED** [review]
Thank you @kanha-gupta, merged.

**Pull request merged**